### PR TITLE
feat: [CDS-36559]: Monitor for zombie workflow execution

### PR DESCRIPTION
**Pull request by Fernando-Dourado**
You can use the following comments to re-trigger PR Checks

- Compile: `trigger compile`
- runAeriformCheck: `trigger AeriformCheck`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/33404)

### CODE REVIEW

**Comment by Fernando-Dourado** [comment]
trigger messagecheck

**Comment by Fernando-Dourado** [comment]
trigger messagecheck

**Comment by Fernando-Dourado** [comment]
trigger checkstyle

File: 400-rest/src/main/java/io/harness/workers/background/critical/iterator/WorkflowExecutionZombieMonitorHandler.java [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+
+        try {
+          workflowExecutionService.triggerExecutionInterrupt(executionInterrupt);
+
+        } catch (InvalidRequestException e) {
+          log.warn(String.format("Unable to honor force abort [workflowId=%s,executionUuid=%s]",
+                       seInstance.getWorkflowId(), seInstance.getExecutionUuid()),
+              e);
+        }
+      }
```
Comment (abhinav-harness) [inline 400-rest/src/main/java/io/harness/workers/background/critical/iterator/WorkflowExecutionZombieMonitorHandler.java:131]: move core logic out of iterator class

File: 400-rest/src/main/java/io/harness/workers/background/critical/iterator/WorkflowExecutionZombieMonitorHandler.java [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+            .entityProcessController(this)
+            .handler(this));
+  }
+
+  /**
+   * The workflow execution should be running and created at least {@code #MAX_RUNNING_MINUTES} ago, then is valid to be
+   * evaluated for zombie conditions.
+   */
+  @Override
+  public boolean shouldProcessEntity(WorkflowExecution entity) {
```
Comment (puthrayaharness) [inline 400-rest/src/main/java/io/harness/workers/background/critical/iterator/WorkflowExecutionZombieMonitorHandler.java:93]: We cannot add this in filter criteria?

File: 400-rest/src/main/java/io/harness/workers/background/critical/iterator/WorkflowExecutionZombieMonitorHandler.java [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+        wfExecution.getWorkflowId());
+
+    List<StateExecutionInstance> stateExecutionInstances =
+        wingsPersistence.createQuery(StateExecutionInstance.class)
+            .filter(StateExecutionInstanceKeys.workflowId, wfExecution.getWorkflowId())
+            .filter(StateExecutionInstanceKeys.executionUuid, wfExecution.getUuid())
+            .asList();
+
+    stateExecutionInstances = CollectionUtils.emptyIfNull(stateExecutionInstances);
+    sort(stateExecutionInstances);
```
Comment (puthrayaharness) [inline 400-rest/src/main/java/io/harness/workers/background/critical/iterator/WorkflowExecutionZombieMonitorHandler.java:109]: Please move all this logic to the DB layer. There is no need to sort and fetch this on the app side. DB can take care of this.

**Review by puthrayaharness — CHANGES_REQUESTED** [review]

**Comment by Fernando-Dourado** [comment]
A wrong rebase create a hard-to-review PR, closing this one and opening another at https://github.com/harness/harness-core/pull/33737.

**Pull request closed without merging**